pkg/app/app.go

```diff
@@ -88,6 +88,19 @@ func defaultFuncApiUrl(config configuration.Configuration, logger *zerolog.Logge
 
 		if len(urlFromOauthToken) > 0 && len(urlFromOauthToken[0]) > 0 {
 			urlString = urlFromOauthToken[0]
+		} else if auth.IsAuthTypePAT(config.GetString(configuration.AUTHENTICATION_TOKEN)) { // derive url from PAT
+			regions := config.GetStringSlice(configuration.SNYK_REGION_URLS)
+			// prefer the existing value
+			if val, ok := existingValue.(string); ok {
+				regions = []string{val}
+			}
+			apiUrl, endpointErr := auth.DeriveEndpointFromPat(config.GetString(configuration.AUTHENTICATION_TOKEN), config, engine.GetNetworkAccess().GetUnauthorizedHttpClient(), regions)
+			if endpointErr != nil {
+				logger.Warn().Err(endpointErr).Msg("failed to get api url from pat")
+			}
+			if len(apiUrl) > 0 {
+				urlString = apiUrl
+			}
 		} else if existingValue != nil { // configured value takes precedence
```

Comment:
thought: I think this comment is not true anymore. It is more the last resort if all else fails.